test/s3.js

```diff
@@ -0,0 +1,191 @@
+/* eslint-env mocha */
+/* global Config */
+'use strict';
+
+const Path = require('path');
+const winston = require('winston');
+const should = require('should');
+const sinon = require('sinon');
+const proxyquire = require('proxyquire');
+
+require('should-sinon');
+
+global.Config = require('../lib/config').load(Path.resolve(__dirname, './data/config.json'));
+
+global.Log = require('../lib/logger').attach(global.Config);
+global.Log.remove(winston.transports.Console);
+
+const NON_DEFAULT_INTERVAL = 10000;
+const DEFAULT_BUCKET = 'fake-bucket';
+
+function generateProxy(stub) {
+  return proxyquire('../lib/source/s3', {
+    'aws-sdk': {
+      S3: function constructor() {
+        return {getObject: stub
+        };
+      }
+    }
+  });
+}
+
+describe('S3 source plugin', () => {
+  let S3,
+      getObjectStub,
+      s3WithNoSuchKeyError,
+      s3WithNotModifiedError,
+      s3OtherError,
+      shutdownSpy;
+
+  beforeEach((done) => {
+    // Stub out all calls to AWS.S3.getObject
+    const fakeResponse = {
+      ETag: 'ThisIsACoolEtag',
+      Body: new Buffer(JSON.stringify({a: 1, b: 'foo', c: {d: 0}}))
+    };
+
+    getObjectStub = sinon.stub().callsArgWith(1, null, fakeResponse);
+    S3 = generateProxy(getObjectStub);
+
+    this.s3 = new S3({bucket: DEFAULT_BUCKET, path: 'foo.json'});
+    done();
+  });
+
+  it('throws an error if instantiated without bucket or path', () => {
+    should.throws(() => {
+      const s3 = new S3(); // eslint-disable-line no-unused-vars
+    }, Error);
+
+    should.throws(() => {
+      const s3 = new S3({  // eslint-disable-line no-unused-vars
```

Comment:
Can you just have `new S3({bucket: DEFAULT_BUCKET});` here and avoid the eslint comment?